specification/sql/resource-manager/Microsoft.Sql/preview/2022-02-01-preview/ManagedInstanceDTCs.json

```diff
@@ -0,0 +1,305 @@
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ManagedInstanceDTCProperties",
+          "description": "Resource properties.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "ManagedInstanceDTCListResult": {
+      "description": "A list of managed instance's DTCs.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Array of results.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ManagedInstanceDTC"
+          },
+          "readOnly": true
+        },
+        "nextLink": {
+          "description": "Link to retrieve next page of results.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "ManagedInstanceDTCProperties": {
+      "description": "The properties of managed instance DTC.",
+      "type": "object",
+      "properties": {
+        "dtcEnabled": {
+          "description": "Active status of managed instance DTC.",
+          "type": "boolean"
+        },
+        "securitySettings": {
+          "$ref": "#/definitions/ManagedInstanceDTCSecuritySettings",
+          "description": "Security settings of managed instance DTC."
+        },
+        "externalDnsSuffixSearchList": {
+          "description": "External dns suffix search list of managed instance DTC.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "dtcHostNameDnsSuffix": {
+          "description": "Host name dns suffix of managed instance DTC.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "ManagedInstanceDTCSecuritySettings": {
+      "description": "The Transaction Manager Communication Settings of the DTC tied to managed instance.",
+      "type": "object",
+      "properties": {
+        "transactionManagerCommunicationSettings": {
+          "$ref": "#/definitions/ManagedInstanceDTCTransactionManagerCommunicationSettings",
+          "description": "Transaction Manager communication settings of managed instance DTC."
+        },
+        "xaTransactionsEnabled": {
+          "description": "Allow XA Transactions to managed instance DTC.",
+          "type": "boolean"
+        },
+        "snaLu6point2TransactionsEnabled": {
+          "description": "Allow SNA LU 6.2 to managed instance DTC.",
+          "type": "boolean"
+        }
+      }
+    },
+    "ManagedInstanceDTCTransactionManagerCommunicationSettings": {
+      "description": "The Transaction Manager Communication Settings of the DTC tied to managed instance.",
+      "type": "object",
+      "properties": {
+        "allowInboundEnabled": {
+          "description": "Allow Inbound traffic to managed instance DTC.",
+          "type": "boolean"
+        },
+        "allowOutboundEnabled": {
+          "description": "Allow Outbound traffic of managed instance DTC.",
+          "type": "boolean"
```

Comment:
consider updating these boolean values to enum as explained here : 

[https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html)

"Replace boolean/switch properties with better enum"

Reply:
Is this really needed? These values should only have True and False as values, and from what I see on the linked post:
"Even if you still believe [True, False] are the correct values for a property, you should use a string enum with values [True, False] instead of boolean. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [True, False, Unknown]."

These values are optional, so not setting them is not a problem (in other words, we don't need the "Unknown" option). Also, since they just determine whether some DTC options are enabled, they can only be "true" or "false". They also cannot be aggregated into a single property since they deal with logically different things.

Reply:
its not mandatory , but in general is a good rule of thumb. Often we see as a service grows , modelling things as boolean puts you in a corner when the scenario scope expands and you end up needing to have another  bool to model something else. In your case it could be something like maybe 3 months down the line you need to have a new state like "allowInboundFromAzureStorage" (hypothetical , I do not know the specifics of your service) and then that ends up becoming another bool rather than an enum which is what it should be.  I will not block sgn off on this comment , but would still would urge you to consider fixing this.